dap_prinz_green_jobs/pipeline/ojo_application/ojo_sample/sample_ojo.py

```diff
@@ -19,25 +23,63 @@ def filter_data(data, filter_ids_set, id_col_name="id"):
 
 
 if __name__ == "__main__":
+    logger.info("Creating and saving job id sample")
+
     deduplicated_ids = pd.read_csv(
         "s3://prinz-green-jobs/outputs/data/ojo_application/deduplicated_sample/deduplicated_job_ids.csv"
     )
+    deduplicated_ids_list = deduplicated_ids.id.to_list()
 
-    # Sample job adverts
+    seed = 42
+    random.seed(seed)
+    sample_ids = random.sample(deduplicated_ids_list, config["ojo_random_sample_size"])
 
-    logger.info("Creating and saving job id sample")
+    logger.info("Loading descriptions data")
 
-    random.seed(42)
-    sample_ids = random.sample(
-        deduplicated_ids["id"].tolist(), config["ojo_random_sample_size"]
+    descriptions_data = pd.read_parquet(config["ojo_s3_file_descriptions"])
+    descriptions_data["id"] = descriptions_data["id"].astype(int)
+    # filter using ALL deuped ids
+    logger.info("Filtering descriptions data on ALL deduplicated ids")
+
+    descriptions_data = descriptions_data[
+        descriptions_data["id"].isin(deduplicated_ids_list)
+    ]
+
+    # sample from the filtered data to get a sample of the descriptions data
+    logger.info("Getting descriptions data sample")
+
+    # use filter data function to get a sample of the descriptions data
+    descriptions_data_sample = filter_data(
+        descriptions_data, set(sample_ids), id_col_name="id"
     )
 
+    # get green jobs here based on keywords
+    logger.info("Getting green descriptions data")
+    descriptions_data_green = descriptions_data[
+        descriptions_data["description"]
+        .str.lower()
+        .str.contains("|".join(green_keywords))
+    ].reset_index(drop=True)
+
+    if len(descriptions_data_green) < config["ojo_random_sample_size"]:
+        sample_green_ids = descriptions_data_green.id.to_list()
+    else:
+        sample_green_ids = random.sample(
+            descriptions_data_green.id.to_list(), config["ojo_random_sample_size"]
+        )
```

Comment:
ok - bit of a curve ball idea here. Maybe we should merge these green ids with an equal number of a random sample of all ids? That way we can just use this one dataset to test everything (and have a nice mix of green and none-green)? We could keep the only-green output files separately too if you think they'd be useful for something by themselves - but maybe output another set of `mixed_` output files too? What do you think?

Reply:
ya sounds like a good idea! will add `mixed_` outfit files too